### ISSUE TITLE
No color on Windows

### ISSUE THREAD

**Issue by muv1** [TwinProduction/go-color#2]
I see that Windows is supported:

<https://github.com/TwinProduction/go-color/blob/master/color_windows.go>

However using this example:

~~~go
package main
import "github.com/TwinProduction/go-color"
func main() {
   println(color.Ize(color.Red, "Something something something"))
   println(color.Red + "Something something something" + color.Reset)
}
~~~

I get no color output.

**Comment by TwinProduction** [comment]
Hey @muv1 

As far as I know, not all versions of Windows support color through the command prompt.
Have a look at https://stackoverflow.com/a/56150498/4603538 or https://stackoverflow.com/a/37783624/4603538

As for the implementation shown in https://github.com/TwinProduction/go-color/blob/master/color_windows.go, it in fact first checks if the command prompt supports color, and if it doesn't, then it replaces all the colors for an empty string.

As cliche as it sounds to say this, it works on my Windows machine: 
![image](https://user-images.githubusercontent.com/15699766/81758875-11eaf100-9491-11ea-8bbc-14d345a8beb1.png)
(I replaced the lower string from red to blue to demonstrate that multiple colors work)

Let me know if the aforementioned solutions fixed it for you.

**Comment by muv1** [comment]
@TwinProduction I dont understand you comment. All versions of Windows support color through the command prompt, going back to at least Windows 7.

The difference being that color via **ANSI escape codes** was only added with Windows 10. Nevertheless, older Windows versions support color via the Windows API. Normally this would be a problem, but it appears youve explictly coded for older Windows support:

https://github.com/TwinProduction/go-color/blob/fa99c9be24b1ac4698d69e580481d80b430e6c23/color_windows.go#L13-L14

So if youve coded for older Windows support, why then go and disable color?

**Comment by TwinProduction** [comment]
@muv1 I think you've misunderstood - 

https://github.com/TwinProduction/go-color/blob/fa99c9be24b1ac4698d69e580481d80b430e6c23/color_windows.go#L13-L16

The above tries to execute `SetConsoleMode` with `0x0001`, `0x0002` and `0x0004`,
If the output is not `The operation completed successfully.`, then the code assumes that the console doesn't support colors via ANSI escape sequences (that you for clarifying that for me), and therefore sets all colors to empty, because otherwise, the ANSI escape sequences would be literally printed.

Have a look at https://github.com/sirupsen/logrus/issues/496

That being said, I dug a little deeper, and it seems like using `ENABLE_VIRTUAL_TERMINAL_INPUT` (0x0200) might be a better candidate.

If you execute the following, without any other dependencies in a new repository and console, what do you get?:
```golang
println("\033[35mHello\033[0m")
```

Once you've run the above, what output do you get from running this:
```golang
var originalMode uint32
handle := windows.Handle(os.Stdout.Fd())
if err := windows.GetConsoleMode(handle, &originalMode); err != nil {
	panic(err.Error())
}
if err := windows.SetConsoleMode(handle, originalMode|windows.ENABLE_VIRTUAL_TERMINAL_PROCESSING); err != nil {
	panic(err.Error())
}
println("\033[35mHello\033[0m")
```

With this information, I should be able to fix the issue.

**Comment by muv1** [comment]
I get:

~~~
undefined: windows
~~~

**Comment by TwinProduction** [comment]
My bad, you need to import `golang.org/x/sys/windows`.

**Comment by muv1** [comment]
~~~
The handle is invalid.
~~~

**Comment by TwinProduction** [comment]
The code I gave you was specifically for cmd, not bash.

What version of Windows do you have, out of curiosity?

**Comment by muv1** [comment]
Windows 7. I should mention before we go any further. This problem has already been solved by another package:

https://github.com/mattn/go-colorable

Maybe you know this already, but if not, some code from that repo could possibly be used to fix this issue.

**Comment by TwinProduction** [comment]
Yeah, I've seen go-colorable before, and it's actually why I made this library.

Executing `go mod vendor` with it, you get 100+ files from its dependencies (mainly due to `golang.org/x/sys`) that try to cover every single use case. 
I like including the vendor folder in my projects, and I wasn't keen in having 8MB of files just to the minority of all use cases.

Like you've pointed out, there are already solutions that support all use cases.

In fact, I pointed this out in the project's README:

> An extremely lightweight cross-platform package to colorize text in terminals.
> This is not meant for maximal compatibility, nor is it meant to handle a plethora of scenarios. It will simply wrap a message with the necessary characters, if the OS handles it.

That being said, I'm not against the idea of supporting more use cases, as long as it doesn't require adding extra dependencies or significantly increasing the size of the project.

**Comment by muv1** [comment]
For something like this:

~~~
package main
import "github.com/TwinProduction/go-color"
func main() {
   println(color.Ize(color.Red, "Something something something"))
}
~~~

Its just a single call to `SetConsoleTextAttribute`. His is more complicated,
as its parsing ANSI escapes into Windows API calls. With yours, the user is
explicitly calling one of your predefined colors, so no parsing needed. Here is
example line:

<https://github.com/mattn/go-colorable/blob/f1b5a0ed/colorable_windows.go#L809>

**Comment by TwinProduction** [comment]
I have a feeling that this is how he's able to get that extra compatibility.

I don't have a Windows 7 machine to try to get it working on without using a similar approach, but if you can give it a try on your end, then I'll be more than happy to review it with you and merge it in this repository.
